### PR TITLE
Improve developer documentation v2

### PR DESCRIPTION
**Pull request by johnnonweiler**
This includes the documentation structure prepared for documentation day (with commits on top of Owen's PR #2732).

### CODE REVIEW

File: doc/architectural/front-end-languages-generating-codet-from-multiple-languages.md [hunk 1]
```diff
@@ -5,24 +5,32 @@
 
 ## language_uit, language_filest, languaget classes: ##
 
-### Purpose ###
-
-To be documented.
+See \ref langapi.
 
 ### Parse ###
```
Comment (peterschrammel) [inline doc/architectural/front-end-languages-generating-codet-from-multiple-languages.md]: Can be removed.

File: doc/architectural/front-end-languages-generating-codet-from-multiple-languages.md [hunk 2]
```diff
@@ -5,24 +5,32 @@
...
 
-To be documented.
+See \ref language_uit::parse().
 
 ### Typecheck ###
```
Comment (peterschrammel) [inline doc/architectural/front-end-languages-generating-codet-from-multiple-languages.md]: Can be removed.

File: doc/architectural/front-end-languages-generating-codet-from-multiple-languages.md [hunk 3]
```diff
@@ -5,24 +5,32 @@
...
 
-To be documented.
+See \ref language_uit::typecheck().
 
 ### Final ###
```
Comment (peterschrammel) [inline doc/architectural/front-end-languages-generating-codet-from-multiple-languages.md]: Can be removed.

File: doc/architectural/front-end-languages-generating-codet-from-multiple-languages.md [hunk 4]
```diff
@@ -5,24 +5,32 @@
...
+
+## C++ ##
+
+See \ref cpp.
 
 ## Java bytecode ##
 
+See \ref java_bytecode.
+
 ### Explain how a java program / class is represented in a .class ###
```
Comment (peterschrammel) [inline doc/architectural/front-end-languages-generating-codet-from-multiple-languages.md]: Should be explained in java_bytecode

File: doc/architectural/solvers-infrastructure.md [hunk 5]
```diff
@@ -3,6 +3,8 @@
 
 \author
 
+See \ref solvers-overview.
+
 ## Flattening ##
```
Comment (peterschrammel) [inline doc/architectural/solvers-infrastructure.md]: Can be removed.

File: src/util/README.md [hunk 6]
```diff
@@ -130,3 +130,17 @@ digraph G {
   2 -> 3 [label="Command line options, file names"];
 }
 \enddot
+
+\subsection ast-examples-section Examples: how to represent the AST of statements, in C and in java
```
Comment (peterschrammel) [inline src/util/README.md:134]: AST-related things might be better located in the context of language frontends (this is langapi, for now).

File: src/solvers/README.md [hunk 7]
```diff
@@ -60,6 +60,20 @@ different decision procedures, roughly one per directory.
 
 * sat/:   Back-ends for a variety of SAT solvers and DIMACS output.
 
+\section solver-apis Solver APIs
+
+\subsection flattening-section Flattening
```
Comment (peterschrammel) [inline src/solvers/README.md]: This is just one instantiation of the SMT API.

File: src/solvers/README.md [hunk 8]
```diff
@@ -60,12 +60,12 @@ different decision procedures, roughly one per directory.
 
 * sat/:   Back-ends for a variety of SAT solvers and DIMACS output.
 
-\section solver-apis Solver APIs
-
-\subsection flattening-section Flattening
+\section flattening-section Flattening
```
Comment (peterschrammel) [inline src/solvers/README.md]: That's an implementation of the SMT API.

**Review by owen-mc-diffblue — APPROVED** [review]

**Pull request merged**